### PR TITLE
Audit event normalization, cleanups, and more

### LINKED ISSUE #63: Update README to be more appropriate for plugins.jenkins.io
The current README is displayed on https://plugins.jenkins.io/audit-log which is a bit too much. This README should be more of an overview of the plugin, and info about contributing can be moved to a new CONTRIBUTING.md file.

### LINKED ISSUE #71: Audit Logs Filling Disk
The audit logs end up filling up all available disk space, can we get some retention configuration options added?

### LINKED ISSUE #73: Audit logs "source_host" identifier
Hi Team,
As a part of extracting data from audit logs, we are looking for a field "source_host" that would give the value of the source Jenkins URL it is pointing to. For e.g, if there are many events recorded for a build trigger, then how the "source_host" field gets the value. Let me know if you could provide further insights on this or anything else required from my end.

### PR DESCRIPTION
**Pull request by jvz**
This fixes #73 by adding the requestURI as a general attribute. This fixes #71 by setting some better default file rollover retention settings. This fixes #63 by cleaning up the readme.

Some other changes:

* Removed the `RequestContext` class as we're not really using it for trace logging type functionality.
* Went through all the existing audit events and made sure they're all specifying required data and other useful normalization.
* Rewrote the log4j2 config from XML to use the programmatic configuration API instead which should make future config updates much easier to implement.
* Update baseline Jenkins requirement to 2.222.4 along with other dependency updates.

### CODE REVIEW

**Comment by jvz** [comment]
Seems as though the tests aren't executing for some reason. I'm fixing the config (and some tests) right now.

**Comment by jeffret-b** [comment]
I'll review once that's straightened out.

**Comment by jvz** [comment]
Yay that fixed the build.

File: src/main/java/io/jenkins/plugins/audit/listeners/UserLogListener.java [hunk 1]
```diff
@@ -22,14 +22,16 @@
...
-import hudson.ExtensionList;
 import hudson.Extension;
+import hudson.ExtensionList;
 import io.jenkins.plugins.audit.event.Login;
 import io.jenkins.plugins.audit.event.Logout;
-import javax.annotation.Nonnull;
 import jenkins.security.SecurityListener;
+import org.apache.logging.log4j.audit.LogEventFactory;
+
+import javax.annotation.Nonnull;
```
Comment (jeffret-b) [inline src/main/java/io/jenkins/plugins/audit/listeners/UserLogListener.java:32]: AuditLogConfiguration uses `edu.umd.cs.findbugs.annotations.NonNull`. Is there a reason for the difference?

File: src/main/java/io/jenkins/plugins/audit/plugins/CustomHTMLLayout.java [hunk 2]
```diff
@@ -14,15 +14,14 @@
  * See the license for the specific language governing permissions and
  * limitations under the license.
  */
-package io.jenkins.plugins.audit;
+package io.jenkins.plugins.audit.plugins;
```
Comment (jeffret-b) [inline src/main/java/io/jenkins/plugins/audit/plugins/CustomHTMLLayout.java:17]: This move / renaming looks odd.

File: src/test/java/io/jenkins/plugins/audit/listeners/NodeChangeListenerTest.java [hunk 3]
```diff
@@ -40,7 +37,7 @@ public void teardown() {
     @Issue("JENKINS-56646")
     @Test
     public void testOnCreated() throws Exception {
-        DumbSlave slave = j.createSlave("TestSlave", "", null);
+        j.createSlave("TestSlave", "", null);
```
Comment (jeffret-b) [inline src/test/java/io/jenkins/plugins/audit/listeners/NodeChangeListenerTest.java:40]:
```suggestion
        j.createSlave("TestAgent", "", null);
```

**Review by jeffret-b — APPROVED** [review]
A nice bit of cleanup and improved behavior.

We should rename to agent whenever possible, such as in the string name in this one test.

File: src/main/java/io/jenkins/plugins/audit/plugins/CustomHTMLLayout.java (hunk 2 above)
Comment (jvz) [reply src/main/java/io/jenkins/plugins/audit/plugins/CustomHTMLLayout.java:17]: The log4j config adds a package to scan for plugins, so to save startup time, I moved the custom plugins into their own package so it has to scan less classes.

File: src/main/java/io/jenkins/plugins/audit/listeners/UserLogListener.java (hunk 1 above)
Comment (jvz) [reply src/main/java/io/jenkins/plugins/audit/listeners/UserLogListener.java:32]: Different class file authors probably?

Comment (jvz) [reply src/main/java/io/jenkins/plugins/audit/listeners/UserLogListener.java]: Updated everything to converge on the findbugs package.

**Comment by jvz** [comment]
CI disk is full at the moment which is causing failures.

**Pull request merged**